packages/fluent-theme/src/private/FluentThemeProvider.tsx

```diff
@@ -52,10 +57,34 @@ const activityMiddleware: readonly ActivityMiddleware[] = Object.freeze([
 
 const sendBoxMiddleware = [() => () => () => PrimarySendBox];
 
+const FluentDecorator = composePipeline<InferDecoratorRequest<typeof ActivityBorderDecoratorRequest>>([
+  function FluentBorderLoader({ request, Next, ...props }) {
+    return (
+      <ActivityLoader showLoader={request.livestreamingState === 'preparing'}>
+        <Next {...props} />
+      </ActivityLoader>
+    );
+  },
+  function FluentBorderFlair({ request, Next, ...props }) {
+    return (
+      <BorderFlair showFlair={request.livestreamingState === 'completing'}>
+        <Next {...props} />
+      </BorderFlair>
+    );
+  }
+]);
+
+FluentDecorator.displayName = 'FluentDecorator';
+
+const FluentDecoratorWithRequest = memo(function DecoratorWithRequest(props) {
+  const request = useDecoratorRequest(ActivityBorderDecoratorRequest);
+  return <FluentDecorator {...props} request={request} />;
+});
+
+FluentDecoratorWithRequest.displayName = 'FluentDecoratorWithRequest';
+
 const decoratorMiddleware: readonly DecoratorMiddleware[] = Object.freeze([
-  createActivityBorderMiddleware(
-    next => request => (request.livestreamingState === 'preparing' ? ActivityLoader : next(request))
-  )
+  createActivityBorderMiddleware(() => () => FluentDecoratorWithRequest)
```

Comment:
This renders into:
![image](https://github.com/user-attachments/assets/323fc721-0784-4af1-ada6-fb5a0d627284)